src/osvr/Common/ImagingComponent.cpp

```diff
@@ -231,12 +230,15 @@ namespace common {
         OSVR_ChannelCount sensor, OSVR_TimeValue const &timestamp) {
 
         auto imageBufferSize = getBufferSize(metadata);
-        auto imageBufferCopy = reinterpret_cast<OSVR_ImageBufferElement*>(cv::fastMalloc(imageBufferSize));
-        memcpy(imageBufferCopy, imageData, imageBufferSize);
+        /// @todo Assuming we should be releasing this pointer from the control
+        /// of the unique_ptr that holds it now, rather than freeing?
+        auto imageBufferCopy = util::makeAlignedImageBuffer(imageBufferSize);
+        memcpy(imageBufferCopy.get(), imageData, imageBufferSize);
 
         Buffer<> buf;
-        messages::ImagePlacedInProcessMemory::MessageSerialization serialization(
-            messages::InProcessMemoryMessage{ metadata, sensor, reinterpret_cast<int>(imageBufferCopy) });
+        messages::ImagePlacedInProcessMemory::MessageSerialization
+            serialization(messages::InProcessMemoryMessage{
+                metadata, sensor, reinterpret_cast<int>(imageBufferCopy)});
```

Comment:
@JeroMiya Would appreciate your help here, I'm sure I'm doing this wrong. The ownership semantics for that buffer aren't terribly clear in this in-process memory implementation, and I'm willing to bet that my guess was wrong, that the usage of `imageBufferCopy` should be `imageBufferCopy.release()` to take control back from the `unique_ptr` that now controls the allocated memory.

Also, `reinterpret_cast<int>(void *)` is terribly unsafe. I know this is for single-process usage, so we don't have to worry about differing pointer sizes on the wire, but at least use an `intptr_t` because `sizeof(int) != sizeof(void*)`